source/api/commands/route2.md

```diff
@@ -0,0 +1,394 @@
+---
+title: route2
+containerClass: experimental
+---
+
+{% note warning %}
+{% fa fa-warning orange %} **This is an experimental feature. In order to use it, you must manually configure the {% url "`experimentalNetworkMocking`" experiments %} option to `true`.** See {% issue 687 %} for more details.
+{% endnote %}
+
+Use `cy.route2()` to manage the behavior of network requests at the network layer.
+
+# Comparison to `cy.route()`
+
+Unlike {% url `cy.route()` route %}, `cy.route2()`:
+
+- can mock all types of network requests including Fetch API, page loads, XMLHttpRequests, etc.
+- does not require calling {% url `cy.server()` server %} before use
+
+# Syntax
+
+```javascript
+// JavaScript
+cy.route2(url)
+cy.route2(url, response)
+cy.route2(method, url)
+cy.route2(method, url, response)
+```
+
+```typescript
+// TypeScript
+cy.route2(url: RouteMatcher, response?: RouteHandler): Chainable<null>
+cy.route2(method: string, url: RouteMatcher, response?: RouteHandler)): Chainable<null>
+```
```

Comment:
@flotwig Good question. So one of the things I wanted to avoid was introducing TypeScript to users who aren't familiar with the syntax. In the new docs, it'd be have more flexibility in terms of letting users choose which one they want to see rather than rendering both.

As far as why there are so many variations, it is to keep consistent with past API signature definitions to show the various ways it could be used.

Reply:
> So one of the things I wanted to avoid was introducing TypeScript to users who aren't familiar with the syntax.

By that logic, let's not have 2 sections for signatures, let's just have 1. That way users don't have to be confused about why there would magically be a different API if they're using Typescript.

> As far as why there are so many variations, it is to keep consistent with past API signature definitions to show the various ways it could be used.

Well, it leaves the `routeMatcher` and `routeHandler` options undocumented in the signatures, so I would say it is failing to show the ways it could be used.

Reply:
@flotwig That's fair. I think given the experimental nature of the feature and what we discussed in Slack. Let's move forward on being a little more progressive with API definitions.